.github/workflows/test-macos.yaml

```diff
@@ -17,18 +17,22 @@ jobs:
       fail-fast: false
       matrix:
         os:
-          - macos-13
+          - macos-15
         ruby:
-          - '2.5'
-          - '2.6'
-          - '2.7'
           - '3.0'
           - '3.1'
           - '3.2'
           - '3.3'
           - '3.4'
           - '3.5'
           # ADD NEW RUBIES HERE
+        include:
+          - os: macos-15-intel
+            ruby: '2.5'
+          - os: macos-15-intel
+            ruby: '2.6'
+          - os: macos-15-intel
+            ruby: '2.7'
```

Comment:
TBH I'd go for removing this -- we still list macOS in https://github.com/DataDog/dd-trace-rb/blob/master/docs/Compatibility.md as "Dev environments only" and I don't think it's very common to run macOS 15 on Intel (and we can always re-add this if we find a use-case)

```suggestion
```